go-app/report/analyzeLog.go

```diff
@@ -134,64 +135,74 @@ func getSpecProcessLogs(cfgFile *Config, fContent string) {
 			}
 			proc_content := proc_rgx_comp.FindAllString(fContent, -1)
 			if len(proc_content) > 1 {
-				FullLogDetails.Analysis_details.SpecificProcess[proc] = strings.Join(proc_content, "\n")
+				mutex.Lock()
+				spec_proc_map[proc] = strings.Join(proc_content, "\n")
+				mutex.Unlock()
 			}
 			waitGroup.Done()
 		}(proc, proc_rgx)
 	}
 	waitGroup.Wait()
-
 }
-func getIssueDetails(cfgFile *Config, fContent string, headerMap map[string]bool) {
-	FullLogDetails.Analysis_details.Issues = make(map[string]map[string]string)
+func getIssueDetails(cfgFile *Config, fContent string, headerMap map[string]bool, issues_map map[string]map[string]string, spec_proc_map map[string]string, grp_issues map[string]GroupedStruct, ngrp_issues map[string]map[string]bool) {
 	specific_proc_content := make(map[string]string)
 	var wg sync.WaitGroup
+	var mutex sync.Mutex
+	var spec_mutex sync.Mutex
 	wg.Add(len(cfgFile.Issues))
-
 	for issue_name, issue := range cfgFile.Issues {
 		go func(issue_name string, issue Issue) {
-			FullLogDetails.Analysis_details.Issues[issue_name] = make(map[string]string)
+			issues_map[issue_name] = make(map[string]string)
 			//Filter the logs belonging to the issue specific process
 			issueContent := ""
 			for proc, proc_rgx := range issue.specific_process {
-				proc_issue, ok := FullLogDetails.Analysis_details.SpecificProcess[proc]
+				proc_issue, ok := spec_proc_map[proc]
 				if !ok {
+					spec_mutex.Lock()
 					proc_issue, ok := specific_proc_content[proc]
```

Comment:
If this is just a local variable scoped within the loop that is in the parallel thread, it shouldn't need a mutex locking writes to it.

Reply:
Okay. Got it